checkstyle/src/main/java/org/infinispan/checkstyle/checks/ForbiddenMethodCheck.java

```diff
@@ -0,0 +1,85 @@
+package org.infinispan.checkstyle.checks;
+
+import static com.puppycrawl.tools.checkstyle.api.TokenTypes.METHOD_CALL;
+
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Check that will log an error when a method that matches the given regular expression is invoked. This method
+ * can optionally on match a method invocation based on the number of arguments provided to it. This check does not
+ * raise errors for test code.
+ */
+public class ForbiddenMethodCheck extends AbstractCheck {
+   private Pattern methodPattern = null;
+   private int argumentCount = -1;
+
+   @Override
+   public int[] getDefaultTokens() {
+      return new int[] { METHOD_CALL };
+   }
+
+   @Override
+   public int[] getAcceptableTokens() {
+      return getDefaultTokens();
+   }
+
+   @Override
+   public int[] getRequiredTokens() {
+      return getDefaultTokens();
+   }
+
+   @Override
+   public void init() {
+      if (methodPattern == null) {
+         throw new IllegalStateException("The methodPattern attribute is required!");
+      }
+   }
+
+   public void setMethodPattern(String methodPatternString) {
+      methodPattern = Pattern.compile(methodPatternString);
+   }
+
+   public void setArgumentCount(String argumentCountString) {
+      argumentCount = Integer.parseInt(argumentCountString);
+      if (argumentCount < 0) {
+         throw new IllegalArgumentException("Argument count " + argumentCount + " was negative");
```

Comment:
Should throw an exception for 0 as well

Reply:
Yeah, I didn't have time to add this yesterday, which is why I just added the comment so I could do it today. Will fix this and document.